File: android/src/main/java/com/beelink/plugins/capacitorzebraprinter/CapacitorZebraPrinterPlugin.java
package com.beelink.plugins.capacitorzebraprinter;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;
import java.io.DataOutputStream;
import java.io.IOException;
import java.net.InetSocketAddress;
import java.net.Socket;
import java.net.SocketAddress;

@CapacitorPlugin(name = "CapacitorZebraPrinter")
public class CapacitorZebraPrinterPlugin extends Plugin {

    @PluginMethod
    public void echo(PluginCall call) {
        String value = call.getString("value");

        JSObject ret = new JSObject();
        ret.put("value", value);
        call.resolve(ret);
    }

    @PluginMethod
    public void print(PluginCall call) {
        String ip = call.getString("ip");
        int port = call.getInt("port");
        String zpl = call.getString("zpl");

        Boolean error = false;
        try {
            Socket clientSocket = null;
            DataOutputStream outToServer = null;
            try {
                System.out.println("Printing now...");
                clientSocket = new Socket();
                SocketAddress socketAddress = new InetSocketAddress(ip, port);
                clientSocket.connect(socketAddress, 30000);
                outToServer = new DataOutputStream(clientSocket.getOutputStream());
                outToServer.writeBytes(zpl);
                System.out.println("send the following string: " + zpl);
            } catch (Exception e) {
                System.out.println("Connection error");
                error = true;
            } finally {
                if (clientSocket != null) {
                    clientSocket.close();
                }
                if (outToServer != null) {
                    outToServer.close();
                }
            }
        } catch (IOException e1) {
            System.out.println("Cannot print label on this printer : " + ip + ":" + port);
            error = true;
        }

        if (error) {
            call.reject("error");
        } else {
            JSObject ret = new JSObject();
            ret.put("value", "success");
            call.resolve(ret);
        }
    }
}

File: ios/Plugin/CapacitorZebraPrinterPlugin.swift
import Foundation
import Capacitor
import SwiftSocket

@objc(CapacitorZebraPrinterPlugin)
public class CapacitorZebraPrinterPlugin: CAPPlugin {
    private let implementation = CapacitorZebraPrinter()

    @objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.resolve([
            "value": value
        ])
    }

    @objc func print(_ call: CAPPluginCall) {
        let ip = call.getString("ip") ?? ""
        let port = call.getInt("port") ?? 0
        let zpl = call.getString("zpl") ?? ""

        let client:TCPClient = TCPClient(address: ip, port: Int32(port))
        switch client.connect(timeout: 30){
            case .success:
                switch client.send(string: zpl) {
                    case .success:
                        guard let data = client.read(1024*10)
                        else {
                            call.resolve([
                                "value": "success"
                            ])
                            return
                        }
                    case .failure(let error):
                        call.reject("error")

                }
            case .failure(let error):
                call.reject("error")
        }
    }
}
